File: src/main/java/fr/epicanard/mapsaver/models/map/ServerMap.java
package fr.epicanard.mapsaver.models.map;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Optional;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Getter
public class ServerMap {
    private int lockedId;
    private Optional<Integer> originalId;
    private String server;
    private UUID mapUuid;
}

File: src/main/scala/fr/epicanard/mapsaver/models/map/ServerMap.scala
package fr.epicanard.mapsaver.models.map

case class ServerMap(
    lockedId: Int,
    originalId: Option[Int],
    world: String,
    x: Int,
    z: Int,
    scale: String,
    server: String,
    dataId: Int
)

object ServerMap {

  def tupled = (ServerMap.apply _).tupled

  def fromMapToSave(mapToSave: MapToSave, dataId: Int, lockedId: Int) =
    new ServerMap(
      lockedId = lockedId,
      originalId = Some(mapToSave.item.id),
      world = mapToSave.item.mapInfo.world,
      x = mapToSave.item.mapInfo.x,
      z = mapToSave.item.mapInfo.z,
      scale = mapToSave.item.mapInfo.scale,
      server = mapToSave.server,
      dataId = dataId
    )
}
